android/src/main/java/com/thunderhead/one_sdk_flutter/OnePlugin.java

```diff
@@ -29,132 +40,271 @@
+        Boolean optOut = (Boolean) arguments.get("optOut");
+        List<String> options = (List<String>) arguments.get("options");
+
+        if (optOut == null) {
+            optOut = false;
+        }
+
+        if (options != null && !options.isEmpty()) {
+            for (String optOutOption : options) {
+                if (optOutOption.equals("cityCountryDetection")) {
+                    Set<OneOptInOptions> optInOptions = EnumSet.noneOf(OneOptInOptions.class);
+                    optInOptions.add(OneOptInOptions.CITY_COUNTRY_DETECTION);
+                    optOutConfiguration = new OneOptOutConfiguration.Builder()
+                            .optOut(optOut)
+                            .optInOptions(optInOptions)
+                            .build();
+                    One.setOptOutConfiguration(optOutConfiguration);
+                }
+                if (optOutOption.equals("allTracking")) {
+                    // Configure all opt-out settings.
+
+                    // Configures opt-out settings for tracking.
+                    optOutConfiguration = new OneOptOutConfiguration.Builder()
+                            .optOut(optOut)
+                            .build();
+                    One.setOptOutConfiguration(optOutConfiguration);
+
+                    // Configures opt-out settings for other options.
+                    Set<OneOptInOptions> allOptInOptions = EnumSet.allOf(OneOptInOptions.class);
+                    optOutConfiguration = new OneOptOutConfiguration.Builder()
+                            .optOut(optOut)
+                            .optInOptions(allOptInOptions)
+                            .build();
+                    One.setOptOutConfiguration(optOutConfiguration);
+                }
+            }
+        } else {
+            // Opt out of everything if no `options` parameter is provided.
+
+            // Configures opt-out settings for tracking.
+            optOutConfiguration = new OneOptOutConfiguration.Builder()
+                    .optOut(optOut)
+                    .build();
+            One.setOptOutConfiguration(optOutConfiguration);
+
+            // Configures opt-out settings for other options.
+            Set<OneOptInOptions> allOptInOptions = EnumSet.allOf(OneOptInOptions.class);
+            optOutConfiguration = new OneOptOutConfiguration.Builder()
+                    .optOut(optOut)
+                    .optInOptions(allOptInOptions)
+                    .build();
+            One.setOptOutConfiguration(optOutConfiguration);
+        }
+        result.success(null);
+    }
+
+    // Helper methods
+
+    // Convert OneResponse Class to HashMap so Dart can read it.
+    private HashMap<String, Object> responseObjectToHashMap(OneResponse response) {
+        HashMap<String, Object> responseMap = new HashMap<>();
+        responseMap.put("tid", response.getTid());
+        responseMap.put("interactionPath", response.getInteractionPath().getValue().getPath());
```

Comment:
why `getPath` ? You should send back the whole uri, not just the path. `android://domain/path` is different from `web://domain/path`. If you just get back the path you can't see that difference.

Reply:
@th-anguyen

Reply:
To  match iOS behavior, as iOS does not give the whole `uri` in the response `interactionPath`

Reply:
If this isn't correct, then we need to create a bug fix ticket for iOS

Reply:
@th-anguyen  Log it and we'll review it in triage. This can proceed.